### ISSUE TITLE
ipa command builds project twice

### ISSUE THREAD

**Issue by stojanovicigi** [KrauseFx/fastlane#106]
I have a lane only with ipa command and my project has been built twice.
Here how lane looks like. I start the build with ```fastlane prod```.
The fastlane version is 0.1.16

```ruby
lane :prod do
  ipa({
      workspace:ENV["WORKSPACE"],
      configuration:"Release",
      scheme: "My Scheme",
      ipa: "my_project.ipa",
      clean: true,
      archive: true,
      verbose:true
    })
end
```

**Comment by KrauseFx** [comment]
Sorry, what is the problem? `fastlane` uses [shenzhen](https://github.com/nomad/shenzhen) under the hood.

**Comment by joshdholtz** [comment]
What does your before block look like? I believe it will build twice if you are also using `xctool`

**Comment by stojanovicigi** [comment]
With ```xctool``` it works good. The block build project only once, as expected.

Here how the block looks like with ```xctool```.

```ruby
lane :prod do  
  # ipa({
  #     workspace:ENV["WORKSPACE"],
  #     configuration:"Release",
  #     scheme: "My Scheme",
  #     ipa: "my_project.ipa",
  #     clean: true,
  #     archive: true,
  #     verbose:true
  #   })

xctool "-workspace /Path/to/CDWorkspace.xcworkspace \
        -scheme 'My Scheme' \
        -configuration Release \
        -sdk iphoneos archive"
end
```

**Comment by KrauseFx** [comment]
Sorry, we need more information to debug your issue here. Please post your full `Fastfile` and full stack trace when running `fastlane`.

**Comment by stojanovicigi** [comment]
You can download or checkout [Fastlane Example](https://bitbucket.org/stojanovicigi/fastlane-example/downloads) project. The project has a script which automatically increase build number.

When you download the project, you should setup the project path in .env file before you run  ```fastlane prod``` command. You'll notice that build number increased by 2.

Looks like the ```ipa``` command builds main target first and increase build number by 1, and then test target and increase build number one more time.

Why it builds test target? Test target shouldn't be built during archiving.
![screen_shot_2015-03-09_at_03_31_09](https://cloud.githubusercontent.com/assets/597534/6549092/1e44c2d0-c60d-11e4-9528-a18da0760ba7.png)

**Comment by stojanovicigi** [comment]
Hi Felix, any news regarding build issue?

**Comment by joshdholtz** [comment]
@stojanovicigi @krausefx I will look into this one today 😊

**Comment by stojanovicigi** [comment]
Thanks :)

**Comment by joshdholtz** [comment]
@stojanovicigi First, thanks for providing an example project :blush: Makes testing a lot easier

Second, I ran `fastlane prod` and saw this output below. There are two lines that have `xcodebuild` in them. Is that what you are seeing that is making it seem like it is building twice? Or are you seeing two `ipa build` commands being called (which I am not seeing)? 
```sh
--- stojanovicigi-fastlane-example-c9c9e237dc14/FastlaneExample » fastlane prod
INFO [2015-03-11 20:28:27.69]: Driving the lane 'prod'
INFO [2015-03-11 20:28:27.69]: -----------------
INFO [2015-03-11 20:28:27.69]: --- Step: ipa ---
INFO [2015-03-11 20:28:27.69]: -----------------
DEBUG [2015-03-11 20:28:27.69]: ipa build -p "../FastlaneExample.xcodeproj" -s "FastlaneExample" -c "Release"
INFO [2015-03-11 20:28:27.69]: [SHELL COMMAND]: ipa build -p "../FastlaneExample.xcodeproj" -s "FastlaneExample" -c "Release"
INFO [2015-03-11 20:28:33.08]: [SHELL OUTPUT]: xcodebuild  ../FastlaneExample.xcodeproj
INFO [2015-03-11 20:28:33.08]: [SHELL OUTPUT]: xcodebuild -sdk iphoneos -project "../FastlaneExample.xcodeproj" -scheme "FastlaneExample" -configuration "Release" clean build archive 1> /dev/null
INFO [2015-03-11 20:28:38.51]: [SHELL OUTPUT]: xcrun  PackageApplication
INFO [2015-03-11 20:28:38.77]: [SHELL OUTPUT]: zip  /Users/josh/Downloads/stojanovicigi-fastlane-example-c9c9e237dc14/FastlaneExample/FastlaneExample.app.dSYM
INFO [2015-03-11 20:28:38.83]: [SHELL OUTPUT]: Successfully built:
INFO [2015-03-11 20:28:38.83]: [SHELL OUTPUT]: /Users/josh/Downloads/stojanovicigi-fastlane-example-c9c9e237dc14/FastlaneExample/FastlaneExample.ipa
INFO [2015-03-11 20:28:38.84]: fastlane.tools finished successfully 🎉
```

**Comment by stojanovicigi** [comment]
@joshdholtz first of all, thank you for your effort.
My fastlane log looks much different then your. You can download the log [from this location](https://bitbucket.org/stojanovicigi/fastlane-example/downloads)

I don't see two ```ipa build``` nor two ```xcodebuild``` commands, but however there are two times
 ```=== BUILD TARGET FastlaneExample OF PROJECT FastlaneExample WITH CONFIGURATION Release ===```

**Comment by KrauseFx** [comment]
I took a quick look and found this in the log.
```
# fastlane 0.2.1 is available. You are on 0.1.16.
```

**Comment by stojanovicigi** [comment]
@KrauseFx Yes, but unfortenully, I have the same issue with version 0.2.1
I attached [new log file](https://bitbucket.org/stojanovicigi/fastlane-example/downloads) with the newest fastlane.

**Comment by joshdholtz** [comment]
@stojanovicigi I believe this is an issue with the configuration of your project. In order to get the output I did before, I had to turn off the running of tests in the scheme (see attached image)

![screen shot 2015-03-12 at 7 17 57 am](https://cloud.githubusercontent.com/assets/401294/6617585/15fb1a66-c888-11e4-835e-b0c98e4969c0.png)

What is also going on is that you have the "Bump Version Number" script in your build phase. Since both of the project and tests get built, the build phase gets ran twice which is why you are seeing it getting incremented twice.

Let me know if I am interpreting that wrong :grin:

**Comment by joshdholtz** [comment]
I get your same response if I call the `xcodebuild` command directly. Not sure exactly what the correct parameters would be though to make that not happen.

**Comment by joshdholtz** [comment]
@stojanovicigi Did you ever get this figured out?

**Comment by stojanovicigi** [comment]
@joshdholtz Not yet, didn't have time to play with it, but I'll try to figure out something during the week.

**Comment by KrauseFx** [comment]
Any updates on this issue?

**Comment by lmirosevic** [comment]
This is actually a problem with Shenzhen. It does a 'clean', then a 'build', then an 'archive'. But of course the 'archive' does a full rebuild so it ends up building your project twice.

I've raised an issue on Shenzhen for this very thing but it hasn't been resolved yet. @stojanovicigi it might help if you confirmed the issue over there: https://github.com/nomad/shenzhen/issues/223

**Comment by KrauseFx** [comment]
@lmirosevic You should probably re-open the issue on `shenzhen` :+1:

**Comment by lmirosevic** [comment]
I was hoping @mattt would reopen it. Didn't want to be too bolshy and create a new issue

**Comment by stojanovicigi** [comment]
@lmirosevic I'll check the issue and confirm it during the day. Thank you guys.

**Comment by eytanbiala** [comment]
@stojanovicigi Were you able to confirm if shenzhen is indeed doing a build and then an archive?

**Comment by KrauseFx** [comment]
As this is an issue of `shenzhen` or `xcodebuild` I'll close this issue for now. Please refer to https://github.com/nomad/shenzhen/issues/223 for further discussion. All `fastlane` does is to trigger `shenzhen` using the `ipa` command.

**Comment by Spencevail** [comment]
Just wanna throw it out there (I'll comment on the `shenzhen` issue as well) that I'm seeing the same issue with my project. I was wondering why this was taking *fooorrreeeevvvverrr*:
**From report.xml**
```
 <testcase name="6: ipa" time="4180.888665"/>
```
